infrastructure/src/main/java/org/corfudb/infrastructure/orchestrator/Orchestrator.java

```diff
@@ -42,7 +39,7 @@
 public class Orchestrator {
 
     final Callable<CorfuRuntime> getRuntime;
-    final Map<String, UUID> activeWorkflows = new ConcurrentHashMap();
+    final Map<UUID, String> activeWorkflows = new ConcurrentHashMap();
```

Comment:
It seems that you need a BiMap instead...

Reply:
Done.